--- a/src/Makefile.am	2025-05-25 13:43:29.363027456 +0000
+++ b/src/Makefile.am	2025-05-25 13:44:38.701129436 +0000
@@ -100,7 +100,7 @@
 	./stage1flex$(EXEEXT) $(AM_LFLAGS) $(LFLAGS) -o $@ $(srcdir)/scan.l
 else
 stage1scan.c: scan.c
-	sed 's|^\(#line .*\)"'`printf %s $< | sed 's|[][\\\\.*]|\\\\&|g'`'"|\1"$@"|g' $< > $@
+	../../.$(HOST_NAME)/src/stage1flex$(EXEEXT) $(AM_LFLAGS) $(LFLAGS) -o $@ $(srcdir)/scan.l
 endif
 
 dist-hook: scan.l flex$(EXEEXT)
