--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -621,7 +621,7 @@
 	set (exe_name "oscam-upx")
 endif (NOT USE_COMPRESS EQUAL 1)
 add_executable (${exe_name} ${exe_srcs} ${exe_hdrs})
-target_link_libraries (${exe_name} ${csoscam} ${csmodules} ${csreaders} csctapi cscrypt minilzo)
+target_link_libraries (${exe_name} ${csoscam} ${csmodules} ${csreaders} csctapi cscrypt minilzo udev)
 if (WITH_SIGNING EQUAL 1)
 	SIGN_COMMAND_OSCAM()
 endif (WITH_SIGNING EQUAL 1)
--- a/utils/CMakeLists.txt
+++ b/utils/CMakeLists.txt
@@ -12,7 +12,7 @@ file (GLOB all_srcs ${exe_srcs})
 
 set (util_name "list_smargo")
 add_executable (${util_name} ${exe_srcs} ${exe_hdrs})
-target_link_libraries (${util_name} ${libusb_link} ${rt_link} ${setupapi_link} ${ole32_link} ${shell32_link} ${pthread_link} ${dl_link})
+target_link_libraries (${util_name} ${libusb_link} ${rt_link} ${setupapi_link} ${ole32_link} ${shell32_link} ${pthread_link} ${dl_link} udev)
 #----------------------- printout resume -----------------------------
 
 message (STATUS "Utils: operating system: ${OSCamOperatingSystem}")
