diff --git a/cmake/deps.cmake b/cmake/deps.cmake
index b63ced21ba89..95498cdb32d5 100644
--- a/cmake/deps.cmake
+++ b/cmake/deps.cmake
@@ -2,14 +2,15 @@ find_library(SONIC_LIB sonic)
 find_path(SONIC_INC "sonic.h")
 find_library(PCAUDIO_LIB pcaudio)
 find_path(PCAUDIO_INC "pcaudiolib/audio.h")
-find_library(PTHREAD_LIB pthread)
+set(THREADS_PREFER_PTHREAD_FLAG ON)
+find_package(Threads)
 find_program(MBROLA_BIN mbrola)
 
 include(FetchContent)
 
-if (PTHREAD_LIB)
+if (CMAKE_USE_PTHREADS_INIT)
   set(HAVE_PTHREAD ON)
-endif(PTHREAD_LIB)
+endif(CMAKE_USE_PTHREADS_INIT)
 if (MBROLA_BIN)
   set(HAVE_MBROLA ON)
 endif(MBROLA_BIN)
diff --git a/src/libespeak-ng/CMakeLists.txt b/src/libespeak-ng/CMakeLists.txt
index dd01aeb2d7c8..bb39dc281ca8 100644
--- a/src/libespeak-ng/CMakeLists.txt
+++ b/src/libespeak-ng/CMakeLists.txt
@@ -109,6 +109,10 @@ if (HAVE_LIBPCAUDIO AND USE_LIBPCAUDIO)
   endif()
 endif()
 
+if (HAVE_PTHREAD)
+  target_link_libraries(espeak-ng PRIVATE Threads::Threads)
+endif()
+
 target_link_libraries(espeak-ng PRIVATE espeak-ng-config ucd)
 if (NOT MSVC)
   target_link_libraries(espeak-ng PRIVATE m)
